src/playbin.js

```diff
@@ -142,22 +162,34 @@ playbin.locked {
         }
     }
 
-    play () {
-        if (!this.uri)
+    get uri () {
+        return this._uri;
+    }
+
+    set uri (value) {
+        if (!value)
             return;
 
         this._ensurePlaybin();
+        this._uri = value;
+        this._playbin.uri = this._uri;
+        this._playbin.set_state(Gst.State.PAUSED);
+    }
+
+    play () {
+        if (!this._uri)
+            return;
 
         this._started = false;
-        this._playbin.uri = this.uri;
+        this._playbin.uri = this._uri;
         this._playbin.set_state(Gst.State.PLAYING);
     }
 
     set locked(value) {
         const style = this.get_style_context();
         if (value)
             style.add_class('locked');
-        else
+        else if (!this.uri)
```

Comment:
Why this check here? I'd rather move this check to whichever caller needed it, and keep this as a simple function.